### ISSUE TITLE
lerna changed should ignore packages where the change should not result in a release.

### ISSUE THREAD

**Issue by jan-molak** [lerna/lerna#2437]
It would be great if `lerna change` could analyse the changes coming in based not only on the files they affected, but also the type of change being introduced.

## Expected Behavior

In a mono-repo using conventional commits, changes marked as `chore`, `test`, `ci`, `docs`, etc. should not result in a package affected being released.

This would be in line with the behaviour of the `semantic-release` module.

## Current Behavior

The [documentation](https://github.com/lerna/lerna/blob/master/commands/changed/README.md) states that `lerna change` supports `ignoreChanges` setting, but this doesn't really help in this case. I would like the package to be released when the `package.json` change is marked as `fix` (say I updated a dep that had a security issue), but not when I've done a `chore` of tweaking an npm script.

## Possible Solution

Look like [`collect-updates`](https://github.com/lerna/lerna/tree/master/utils/collect-updates) would have to be changed to recognise the commit messages that affected the files.

## Steps to Reproduce (for bugs)

1. Create a mono-repo managed by lerna
2. Add dependabot service
3. See dependabot update your dev dependencies
4. See lerna publish all the packages to npm even though they have only received `chore` updates to their `devDependencies`

<details><summary>lerna.json</summary><p>

```json
{
  "lerna": "2.11.0",
  "loglevel": "info",
  "packages": [
    "documentation/*",
    "examples/*",
    "integration/*",
    "packages/*"
  ],
  "version": "2.0.8",
  "command": {
    "create": {
      "homepage": "https://serenity-js.org",
      "license": "Apache-2.0"
    },
    "version": {
      "allowBranch": "master",
      "conventionalCommits": true,
      "exact": true,
      "message": "chore(release): %s\n[ci skip]"
    },
    "ignoreChanges": [
      "**/documentation/**",
      "**/examples/**",
      "**/integration/**",
      "**/*.md",
      "**/spec/**"
    ],
    "publish": {
      "dist-tag": "latest"
    }
  }
}
```

Sadly, adding `ignoreChange` doesn't work either as changes introduced in the ignored directories still trigger a release. That's probably an issue for another ticket though.

</p></details>

## Context

I'm trying to have lerna automatically version and publish the public packages at https://gihub.com/serenity-js/serenity-js without bumping the version numbers, polluting the changelog, and releasing the packages when it's not necessary.

## Your Environment

Please have a look at the repository at https://github.com/serenity-js/serenity-js

| Executable | Version |
| ---: | :--- |
| `lerna --version` | 3.20.2 |
| `npm --version`  | 6.13.4 |
| `yarn --version` | N/A |
| `node --version` | 10.18.1 |

| OS | Version |
| --- | --- |
| NAME | VERSION |
| Travis CI | travis-ci-sardonyx-xenial |

**Comment by joelmukuthu** [comment]
Hi @jan-molak, @evocateur has mentioned that this will never happen in lerna. See https://github.com/lerna/lerna/issues/1569#issuecomment-413376084 and https://github.com/lerna/lerna/issues/1569#issuecomment-413743100. So this will have to be handled in semantic-release.

(_I'm only a messenger_)

**Comment by jan-molak** [comment]
Thanks @joelahoover, I didn't realise that this was the case. I simply assumed that respecting semantic release conventions would be a sensible thing for Lerna to do, hence this ticket.

I'm not entirely sure what @evocateur refers to in [#1569 (comment)](https://github.com/lerna/lerna/issues/1569#issuecomment-414096502) as "the fundamental contract of lerna", so would love some more clarity on that.

I suppose another alternative would be to write some sort of a script, as per @hutson's [comment](https://github.com/lerna/lerna/issues/1569#issuecomment-414094159), that analysed the commits since the last release and invoked `lerna publish` only if they contained any patches/features/breaking changes, etc; or monkey-patch Lerna's [collect-updates](https://github.com/lerna/lerna/blob/6c4ee5294002c0e464f557b149516017eccf2a07/utils/collect-updates/collect-updates.js). 

Either approach smells of a hack, though, and I'd be much more keen for Lerna to enable some way for the user to configure the tool's behaviour. I wouldn't of course expect lerna to cater for every single possible use case, simply provide an extension point that would allow the user to inject whatever strategy/plugin is appropriate for their monorepo.

**Comment by joelmukuthu** [comment]
Yeah, all workarounds. Even the yarn 2 approach would need something to analyse commits and generate deferred versions. 

Ideally, Lerna would allow collect-versions to be pluggable (I don't know if this already possible) or semantic release would know how to deal with monorepos. I think it's easier handled in semantic-release, since it already supports plugins and this issue https://github.com/atlassian/lerna-semantic-release/issues/89 suggests that it could already be done.

**Comment by jan-molak** [comment]
Yeah, that's what I'm hoping for as well; support for monorepos has been discussed for quite some time under semantic-release/semantic-release#193

**Comment by blake-newman** [comment]
@jan-molak we also find the same issue; maintaining a large monorepo with core tooling for our company is a bit hectic at the moment because of this reason.

We want to automate the workflow of publishing, however, the lack of being able to only publish when there are changes via the semantic versioning spec means there is a publish for packages even for something small like a dev dependency update. This means we are publishing many many packages just to keep our dependencies up to date. 

The alternative currently is to manually release at the desired time however we want to avoid that as when is the 'right' time. 

It would be hugely beneficial if there was the ability to filter based on commit type so only particular commit types would bump packages.

**Comment by achingbrain** [comment]
This does seem like an oversight - doing things like updating dev deps should not result in a patch release for a module.

**Comment by mohanraj-r** [comment]
On a related note want to bring up the discrepancy between `lerna changed` and `lerna list`.

While `lerna list --since master` e.g. is able to detect changes correctly and list packages accordingly, `lerna changed` neither supports the `--since` arg nor does the change detection correctly as `lerna list`. With `lerna changed` we get `lerna info Assuming all packages changed`. And `lerna version` seems to rely/model on `lerna changed` behavior rather than `lerna list --since ..`. Is there a reason behind this? Could this be fixed e.g. to add `--since` behavior to `lerna version` similar to `lerna list`?
